include/GafferUSD/USDLayerWriter.h

```diff
@@ -0,0 +1,98 @@
+//////////////////////////////////////////////////////////////////////////
+//
+//  Copyright (c) 2022, Cinesite VFX Ltd. All rights reserved.
+//
+//  Redistribution and use in source and binary forms, with or without
+//  modification, are permitted provided that the following conditions are
+//  met:
+//
+//      * Redistributions of source code must retain the above
+//        copyright notice, this list of conditions and the following
+//        disclaimer.
+//
+//      * Redistributions in binary form must reproduce the above
+//        copyright notice, this list of conditions and the following
+//        disclaimer in the documentation and/or other materials provided with
+//        the distribution.
+//
+//      * Neither the name of John Haddon nor the names of
+//        any other contributors to this software may be used to endorse or
+//        promote products derived from this software without specific prior
+//        written permission.
+//
+//  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS
+//  IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+//  THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+//  PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+//  CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+//  EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+//  PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+//  PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+//  LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+//  NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+//  SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+//////////////////////////////////////////////////////////////////////////
+
+#ifndef GAFFERUSD_USDLAYERWRITER_H
+#define GAFFERUSD_USDLAYERWRITER_H
+
+#include "GafferUSD/Export.h"
+#include "GafferUSD/TypeIds.h"
+
+#include "GafferScene/ScenePlug.h"
+#include "GafferScene/SceneWriter.h"
+
+#include "GafferDispatch/TaskNode.h"
+
+#include "Gaffer/TypedPlug.h"
+#include "Gaffer/StringPlug.h"
+
+namespace GafferUSD
+{
+
+class GAFFERUSD_API USDLayerWriter : public GafferDispatch::TaskNode
```

Comment:
I don't think we reached any real conclusion on the name this morning.  "USD" "Writer" and "Layer" do all seem like fairly important qualifiers, but I do wish we could fit "Diff" in there somewhere as well.

Reply:
Yeah, that's fine.  At least it has "USD" in the name, which is a good signpost for "better double check what it actually is because the naming probably won't be clear" :P